youtube_transcript_api/_transcripts.py

```diff
@@ -0,0 +1,202 @@
+            if '"playabilityStatus":' not in html:
+                raise VideoUnavailable(video_id)
+
+            raise TranscriptsDisabled(video_id)
+
+        return json.loads(splitted_html[1].split(',"videoDetails')[0].replace('\n', ''))[
+            'playerCaptionsTracklistRenderer'
+        ]
+
+    def _fetch_html(self, video_id):
+        return self._http_client.get(WATCH_URL.format(video_id=video_id)).text.replace(
+            '\\u0026', '&'
+        ).replace(
+            '\\', ''
+        )
+
+
+class TranscriptData():
+    # TODO implement iterator
+
+    def __init__(
+        self, http_client, video_id, manually_created_transcripts, generated_transcripts, translation_languages
+    ):
+        self._http_client = http_client
+        self.video_id = video_id
+        self._manually_created_transcripts = manually_created_transcripts
+        self._generated_transcripts = generated_transcripts
+        self._translation_languages = translation_languages
+
+    @staticmethod
+    def build(http_client, video_id, captions_json):
+        manually_created_transcripts = []
+        generated_transcripts = []
+
+        for caption in captions_json['captionTracks']:
+            (generated_transcripts if caption.get('kind', '') == 'asr' else generated_transcripts).append(
+                {
+                    'url': caption['baseUrl'],
+                    'language': caption['name']['simpleText'],
+                    'language_code': caption['languageCode'],
+                    'is_generated': caption.get('kind', '') == 'asr',
+                    'is_translatable': caption['isTranslatable'],
+                }
+            )
+
+        return TranscriptData(
+            http_client,
+            video_id,
+            manually_created_transcripts,
+            generated_transcripts,
+            [
+                {
+                    'language': translation_language['languageName']['simpleText'],
+                    'language_code': translation_language['languageCode'],
+                } for translation_language in captions_json['translationLanguages']
+            ],
+        )
+
+    def find_transcript(self, language_codes):
+        try:
+            return self.find_manually_created_transcript(language_codes)
+        except NoTranscriptFound:
+            pass
+
+        return self.find_generated_transcript(language_codes)
+
+    def find_generated_transcript(self, language_codes):
+        return self._find_transcript(language_codes, generated=True)
+
+    def find_manually_created_transcript(self, language_codes):
+        return self._find_transcript(language_codes, generated=False)
+
+    def _find_transcript(self, language_codes, generated):
+        transcripts = self._generated_transcripts if generated else self._manually_created_transcripts
+
+        for language_code in language_codes:
+            for transcript in transcripts:
+                if transcript['language_code'] == language_code:
```

Comment:
I thought about that. However, that would imply a few other things, as language codes can return multiple results. Since it is now possible to list all available transcripts with their corresponding language codes, the users could implement this behavior themselves if that's what they desire. If we allow a language code matching multiple language codes, we might get a less desirable result by default, which the user can't work around as easily.